autopilot/hostfilter.go

```diff
@@ -300,7 +307,7 @@ func isOutOfCollateral(c api.Contract, s rhpv2.HostSettings, renterFunds types.C
 // our expectation using the host settings.
 func isBelowCollateralThreshold(expectedCollateral, actualCollateral types.Currency) bool {
 	if expectedCollateral.IsZero() {
-		return true // protect against division-by-zero
+		return false // protect against division-by-zero
```

Comment:
This was revealed by `TestSectorPruning` failing non-stop after capping the storage amount. Because it actually fills up a host, causing this to be 0 and triggering a refresh that we didn't need which changes the contract id mid-test. That leads to the test failing because the contract to prune isn't found anymore.

Reply:
🕵️ (edit: maybe this warrants a comment explaining when expected collateral is zero)